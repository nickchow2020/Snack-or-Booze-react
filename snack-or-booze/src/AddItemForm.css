.onSubmitForm{
    width: 40%;
}

.btnBack{
    margin-left: 1rem;
}