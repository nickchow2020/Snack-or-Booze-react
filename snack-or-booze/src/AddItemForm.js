import React,{useState} from 'react';
import { useHistory,Link} from 'react-router-dom';
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';
import "./AddItemForm.css"
const AddItemForm = ({add})=>{

    const history = useHistory()

    const INITIAL_VALUE = {
        id:"",
        name:"",
        description:"",
        recipe:"",
        serve:""
    }

    const [data,setData] = useState(INITIAL_VALUE)

    const convertId = (name)=>{
        return name.replaceAll(" ","-")
    }

    const handleChange = (evt)=>{   
        const {name,value} = evt.target;

        setData(prevData => ({
            ...prevData,
            [name]:value,
        }))
    }

    const handleSubmit = ()=>{
        const snack = {
            ...data,
            id:convertId(data.name)
        }
        add(snack)
        setData(INITIAL_VALUE)
        history.push("/snacks")
    }

    return (
        <Form onSubmit={handleSubmit} className="onSubmitForm">
            <h1>Add New Snack</h1>
            <FormGroup>
                <Label for="name">Name</Label>
                <Input 
                type="text" 
                name="name" 
                id="name" 
                placeholder="enter item name" 
                value={data.name} 
                onChange={handleChange} />
            </FormGroup>
            <FormGroup>
                <Label for="description">Description</Label>
                <Input 
                type="text" 
                name="description" 
                id="description" 
                placeholder="description" 
                value={data.description} 
                onChange={handleChange}/>
            </FormGroup>
            <FormGroup>
                <Label for="recipe">Recipe</Label>
                <Input 
                type="text" 
                name="recipe" 
                id="recipe" 
                placeholder="recipe" 
                value={data.recipe} 
                onChange={handleChange} />
            </FormGroup>
            <FormGroup>
                <Label for="serve">Serve</Label>
                <Input 
                type="text" 
                name="serve" 
                id="serve" 
                placeholder="serve" 
                value={data.serve} 
                onChange={handleChange} />
            </FormGroup>
            <Button>Add New Item</Button>
            <Link to="/snacks"><Button className="btnBack">Go Back</Button></Link>
        </Form>
    )
}

export default AddItemForm