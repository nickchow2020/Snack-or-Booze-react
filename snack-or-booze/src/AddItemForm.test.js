import React from "react";
import {fireEvent, render} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import {MemoryRouter} from "react-router-dom";
import AddItemForm from "./AddItemForm"
it("should it render",function(){
    render(
    <MemoryRouter>
        <AddItemForm />
    </MemoryRouter>
    )
}) 

it("should it match Snapshot",function(){
    const {asFragment} = render(
        <MemoryRouter initialEntries={["/snacks/add"]}>
            <AddItemForm />
        </MemoryRouter>
        )

    expect(asFragment()).toMatchSnapshot()
})


it("should it match context",function(){
    const {getByText,getByPlaceholderText} = render(
        <MemoryRouter initialEntries={["/snacks/add"]}>
            <AddItemForm />
        </MemoryRouter>
        )

    expect(getByText("Add New Snack")).toBeInTheDocument()
    expect(getByPlaceholderText("enter item name")).toBeInTheDocument()
})
