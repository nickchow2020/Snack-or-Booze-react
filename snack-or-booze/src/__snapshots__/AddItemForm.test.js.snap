// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should it match Snapshot 1`] = `
<DocumentFragment>
  <form
    class="onSubmitForm"
  >
    <h1>
      Add New Snack
    </h1>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="name"
      >
        Name
      </label>
      <input
        class="form-control"
        id="name"
        name="name"
        placeholder="enter item name"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="description"
      >
        Description
      </label>
      <input
        class="form-control"
        id="description"
        name="description"
        placeholder="description"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="recipe"
      >
        Recipe
      </label>
      <input
        class="form-control"
        id="recipe"
        name="recipe"
        placeholder="recipe"
        type="text"
        value=""
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <label
        class=""
        for="serve"
      >
        Serve
      </label>
      <input
        class="form-control"
        id="serve"
        name="serve"
        placeholder="serve"
        type="text"
        value=""
      />
    </div>
    <button
      class="btn btn-secondary"
    >
      Add New Item
    </button>
    <a
      href="/snacks"
    >
      <button
        class="btnBack btn btn-secondary"
      >
        Go Back
      </button>
    </a>
  </form>
</DocumentFragment>
`;
